lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -228,6 +228,37 @@ class DecomposeAtenSelectIntOp : public OpRewritePattern<AtenSelectIntOp> {
 };
 } // namespace
 
+namespace {
+class DecomposeAtenNarrowOp : public OpRewritePattern<AtenNarrowOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenNarrowOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Location loc = op.getLoc();
+    Value start = op.start();
+    Value dim = op.dim();
+    Value length = op.length();
+
+    Value one =
+        rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(1));
+    Value startPlusLength =
+        rewriter.create<AtenAddIntOp>(loc, one.getType(), start, length);
+
+    // `aten.slice.tensor` doesn't squeeze the dim even when it's size 1 after
+    // slicing, while `aten.select.int` does.
+
+    // Replaces the AtenSliceTensorOp op, provides values from the nearly
+    // identical aten::narrow
+    rewriter.replaceOpWithNewOp<AtenSliceTensorOp>(
+        op, op.getResult().getType(), op.self(), /*dim*/ dim, /*start*/ start,
+        /*end*/ startPlusLength, /*step=*/one);
```

Comment:
the convention for adding the arg name as a comment is `/*argname=*/variable`. Some of the comments here are missing the `=` and there is an extra whitespace after the comments